addon/mixins/nested-relations.js

```diff
@@ -203,7 +214,9 @@ export default Mixin.create({
 
       let adapterOptions = snapshot.adapterOptions || {};
 
-      let attributes = attributesFor(snapshot.record);
+      let relationships = relationshipsDirective(adapterOptions.relationships);
+
+      let attributes = attributesFor(snapshot.record, relationships);
```

Comment:
`attributesFor` is getting only the subsets of relationships that are passed in for serialization. This means the `attributes` can only contain relationship ids of things that are normally serialized, so `(1)` from https://github.com/jsonapi-suite/ember-data-extensions/pull/23#issue-331124621 cannot be done.

Reply:
We should not be discluding from relationships within this method. It should be done within the method that serializes relationships